internal/controller/notebook_controller.go

```diff
@@ -48,9 +51,25 @@ func (r *NotebookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	}
 
 	if result, err := r.reconcileNotebook(ctx, &notebook); !result.success {
+		log.Info("success: %b", result.success)
 		return result.Result, err
 	}
 
+	if (notebook.Spec.Upload.Md5Checksum != "" && notebook.Status.UploadURL == "") ||
```

Comment:
I was envisioning this going under `.spec.image.upload` which would then move this to `ContainerImageReconciler`

Reply:
I had it stuck in my head that upload was top level but the design says otherwise.